### PR TITLE
Mock: SharePoint: List objects expansion

### PR DESCRIPTION
**Pull request by dadams39**
## Description
Adds the ability to create non-empty lists within mocks

## Does this PR need a docs update or release note?

- [x] :no_entry: No 

## Type of change

- [x] :robot: Test
- [x] :computer: CI/Deployment

## Issue(s)

*closes to  #2018<issue>

## Test Plan

- [x] :zap: Unit test

### CODE REVIEW

**Comment by dadams39** [comment]
Need to rebase prior to review

**Review by ryanfkeepers — APPROVED** [review]

File: src/internal/connector/mockconnector/mock_data_list.go [hunk 1]
```diff
@@ -139,3 +169,103 @@ func GetMockListStream(t *testing.T, title string) *MockListData {
 
 	return listData
 }
+
+// getItems returns a map where key values are albums
+// and values are the artist.
+// Source: https://github.com/Currie32/500-Greatest-Albums/blob/master/albumlist.csv
+func getItems() map[string]string {
```
[the remark this answers is not in the document]
Comment (dadams39) [reply src/internal/connector/mockconnector/mock_data_list.go:176]: LoL, my 5 favorites are not here. I will drop the list and add a wrapper.

File: src/internal/connector/mockconnector/mock_data_list.go [hunk 2]
```diff
@@ -75,30 +77,24 @@ func (mld *MockListData) ToReader() io.ReadCloser {
 }
 
 // GetMockList returns a Listable object with generic
-// information.
+// information. The max amount of unique values is:
```
[the remark this answers is not in the document]
Comment (dadams39) [reply src/internal/connector/mockconnector/mock_data_list.go:80]: Will adjust the number.

**Comment by dadams39** [comment]
Local tests pass
```bash
$ go test ./...
ok  	github.com/alcionai/corso/src/internal/connector	273.146s
ok  	github.com/alcionai/corso/src/internal/connector/discovery	0.231s
ok  	github.com/alcionai/corso/src/internal/connector/exchange	109.106s
ok  	github.com/alcionai/corso/src/internal/connector/graph	0.248s
ok  	github.com/alcionai/corso/src/internal/connector/mockconnector	0.455s
ok  	github.com/alcionai/corso/src/internal/connector/onedrive	12.729s
ok  	github.com/alcionai/corso/src/internal/connector/sharepoint	7.869s
ok  	github.com/alcionai/corso/src/internal/connector/support	0.265s
ok  	github.com/alcionai/corso/src/internal/connector/uploadsession	0.453s
```

File: src/internal/connector/mockconnector/mock_data_list.go [hunk 3]
```diff
@@ -107,12 +103,46 @@ func GetMockList(title string) models.Listable {
...
+		fields.SetAdditionalData(entry)
+		temp.SetFields(fields)
+
+		itms = append(itms, temp)
+		index++
+
+		if index == count {
+			break
+		}
+	}
```
[the remark this answers is not in the document]
Comment (dadams39) [reply src/internal/connector/mockconnector/mock_data_list.go]: A version of changes applied within [adaa55c](https://github.com/alcionai/corso/pull/2019/commits/adaa55c12625972463bfd7c7b3b7c9fab957e154)

File: src/internal/connector/mockconnector/mock_data_list.go [hunk 4]
```diff
@@ -75,30 +77,24 @@ func (mld *MockListData) ToReader() io.ReadCloser {
 }
 
 // GetMockList returns a Listable object with generic
-// information.
+// information. The max amount of unique values is:
+// If count exceeds max amount, the returned list will not be able to be loaded into
+// the M365 back store.
 // Source: https://learn.microsoft.com/en-us/graph/api/list-create?view=graph-rest-1.0&tabs=go
-func GetMockList(title string) models.Listable {
+func GetMockList(title string, count int) models.Listable {
```
[the remark this answers is not in the document]
Comment (dadams39) [reply src/internal/connector/mockconnector/mock_data_list.go]: Changed to map input for Generic use case. Created also default use case within [adaa55c](https://github.com/alcionai/corso/pull/2019/commits/adaa55c12625972463bfd7c7b3b7c9fab957e154)

File: src/internal/connector/mockconnector/mock_data_list.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dadams39) [reply src/internal/connector/mockconnector/mock_data_list.go:167]: List reduced to 20 within [adaa55c](https://github.com/alcionai/corso/pull/2019/commits/adaa55c12625972463bfd7c7b3b7c9fab957e154)

**Pull request merged**